src/reducer/auth-reducer.js

```
@@ -0,0 +1,30 @@
+const authReducer = (state, action) => {
+  switch (action.type) {
+    case "LOGIN":
+      return {
+        ...state,
+        user: action.payload.user,
+        token: action.payload.token,
+      };
+
+    case "SIGNUP":
+      return {
+        ...state,
+        user: action.payload.user,
+        token: action.payload.token,
+      };
+
+    case "CHECK_USER":
+      return {
+        ...state,
+        user: action.payload.user,
+        token: action.payload.token,
+      };
+    case "LOGOUT":
+      return { ...state, user: null, payload: null };
+    default:
+      return state;
```

You can also throw an Error object here instead of returning the state.
But for now, this also works.

---

Error object as in? 
Can you explain it a bit more?

---

You can read more on Error objects.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error